staging/src/k8s.io/apimachinery/pkg/runtime/serializer/allocator.go

```diff
@@ -0,0 +1,42 @@
+package serializer
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// AllocatorPool simply stores Allocator objects to avoid additional memory allocations
+// by caching created but unused items for later reuse, relieving pressure on the garbage collector.
+//
+// Usage:
+//  memoryAllocator := serializer.AllocatorPool.Get().(*serializer.Allocator)
+//  defer serializer.AllocatorPool.Put(memoryAllocator)
+var AllocatorPool = sync.Pool{
+	New: func() interface{} {
+		return &Allocator{}
+	},
```

Comment:
One more thing I was thinking about is whether we want a single pool, or maybe we want to go one step further (e.g. buffers of size 1000, 10000, 100000 and 10000000 (or whatever)). 
This could allow us to be slightly more efficient, but maybe I'm actually overengineering it and what you're proposing here is actually enough.

Reply:
it is hard to say, I think we would have to benchmark it to see the value. I think we could start with the simple pool and refine it in the future.

Reply:
I'm fine with starting simple - just maybe add a comment that something like that may be considered later.